src/algokit_utils/models/amount.py

```diff
@@ -0,0 +1,196 @@
+from __future__ import annotations
+
+import algosdk
+from typing_extensions import Self
+
+__all__ = ["AlgoAmount"]
+
+
+class AlgoAmount:
+    """Wrapper class to ensure safe, explicit conversion between µAlgo, Algo and numbers.
+
+    :param amount: A dictionary containing either algos, algo, microAlgos, or microAlgo as key
+                    and their corresponding value as an integer or Decimal.
+    :raises ValueError: If an invalid amount format is provided.
+
+    :example:
+    >>> amount = AlgoAmount({"algos": 1})
+    >>> amount = AlgoAmount({"microAlgos": 1_000_000})
+    """
+
+    def __init__(self, amount: dict[str, int]):
```

Comment:
```suggestion
    def __init__(self, amount: dict[str, int | Decimal]):
```
If it's algo it can be a decimal value.